src/main/java/seedu/ddd/logic/parser/ParserUtil.java

```diff
@@ -223,4 +231,69 @@ public static CommandFlag parseFlags(ArgumentMultimap argMultimap) throws ParseE
             return null;
         }
     }
+
+    /**
+     * Verifies that the specified ArgumentMultimap does not contain any illegal filters for a client.
+     * Throws a ParseException if a forbidden filter (service, description, or date) is found when
+     * the client flag (-c) is specified.
+     *
+     * @param argMultimap The ArgumentMultimap containing the parsed arguments.
+     * @throws ParseException if the ArgumentMultimap contains filters for service, description, or date
+     *         when the client flag (-c) is specified.
+     */
+    public static void verifyClientParser(ArgumentMultimap argMultimap) throws ParseException {
+        if (argMultimap.getValue(PREFIX_SERVICE).isPresent()) {
+            throw new ParseException(getErrorMessageForPrefix(PREFIX_SERVICE, FLAG_CLIENT));
+        } else if (argMultimap.getValue(PREFIX_DESC).isPresent()) {
+            throw new ParseException(getErrorMessageForPrefix(PREFIX_DESC, FLAG_CLIENT));
+        } else if (argMultimap.getValue(PREFIX_DATE).isPresent()) {
+            throw new ParseException(getErrorMessageForPrefix(PREFIX_DATE, FLAG_CLIENT));
+        } else {
+
+        }
+    }
+
+    /**
+     * Verifies that the specified ArgumentMultimap does not contain any illegal filters for a vendor.
+     * Throws a ParseException if a forbidden filter (description or date) is found when
+     * the vendor flag (-v) is specified.
+     *
+     * @param argMultimap The ArgumentMultimap containing the parsed arguments.
+     * @throws ParseException if the ArgumentMultimap contains filters for description or date
+     *         when the vendor flag (-v) is specified.
+     */
+    public static void verifyVendorParser(ArgumentMultimap argMultimap) throws ParseException {
+        if (argMultimap.getValue(PREFIX_DESC).isPresent()) {
+            throw new ParseException(getErrorMessageForPrefix(PREFIX_DESC, FLAG_VENDOR));
+        } else if (argMultimap.getValue(PREFIX_DATE).isPresent()) {
+            throw new ParseException(getErrorMessageForPrefix(PREFIX_DATE, FLAG_VENDOR));
+        } else {
+
```

Comment:
I think you can either leave a comment here or just omit the last `else` block instead of having an empty `else` block.

Reply:
Updated with a comment. If I omit the last `else` block, it will fail the checkstyle script.